services/src/main/java/org/keycloak/services/resources/admin/RealmsAdminResource.java

```diff
@@ -136,8 +136,14 @@ public Response importRealm(final RealmRepresentation rep) {
         logger.debugv("importRealm: {0}", rep.getRealm());
 
         try {
-            RealmModel realm = realmManager.importRealm(rep);
-            grantPermissionsToRealmCreator(realm);
+            RealmModel realm = realmManager.getRealm(rep.getId());
+            if (realm == null) {
+                realm = realmManager.getRealmByName(rep.getRealm());
+            }
+            if (realm == null) {
```

Comment:
if realm is not null (i.e. a realm with the same id or the same name exists in the DB), it will not be imported.
I think this deserves a warning trace instead letting one thinks that the realm was actually imported